pandas/core/frame_test_constructors.py

```diff
@@ -0,0 +1,7 @@
+import pandas as pd
+def test_empty_df_preserve_col():
+    rows = []
+    df = pd.DataFrame.from_records(iter(rows), columns=['col_1', 'Col_2'], nrows=0)
+    assert list(df.columns)==['col_1', 'Col_2']
+    assert len(df) == 0
```

Comment:
Can you follow the dev docs here: https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests

Namely, search the current tests for `from_records` and that should give a good indication of where to place this test.

Reply:
This file needs to be removed.